core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/ReportEditViewModel.kt

```diff
@@ -0,0 +1,230 @@
+package com.ustadmobile.core.viewmodel.report
+
+import com.ustadmobile.core.MR
+import com.ustadmobile.core.domain.report.model.ReportFilter3
+import com.ustadmobile.core.domain.report.model.ReportOptions2
+import com.ustadmobile.core.domain.report.model.ReportSeries2
+import com.ustadmobile.core.impl.appstate.ActionBarButtonUiState
+import com.ustadmobile.core.impl.appstate.AppUiState
+import com.ustadmobile.core.impl.appstate.LoadingUiState
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.ext.replace
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.ReportFilterEditViewModel
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.door.ext.withDoorTransactionAsync
+import com.ustadmobile.lib.db.entities.Report
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.kodein.di.DI
+
+class ReportEditViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+    destName: String = ""
+) : UstadEditViewModel(di, savedStateHandle, DEST_NAME) {
+
+    private val _uiState: MutableStateFlow<ReportEditUiState> =
+        MutableStateFlow(ReportEditUiState())
+    val uiState: Flow<ReportEditUiState> = _uiState.asStateFlow()
+    private val entityUid: Long
+        get() = savedStateHandle[UstadView.ARG_ENTITY_UID]?.toLong() ?: 0
+
+    init {
+        loadingState = LoadingUiState.INDETERMINATE
+        val title = systemImpl.getString(MR.strings.edit_report)
+
+        _appUiState.update {
+            AppUiState(
+                title = title,
+                hideBottomNavigation = false
+            )
+        }
+        launchIfHasPermission(
+            setLoadingState = true,
+            permissionCheck = { true }
+        ) {
+            async {
+                loadEntity(
+                    serializer = Report.serializer(),
+                    onLoadFromDb = { db ->
+                        db.reportDao().findByUid(entityUid)
+                    },
+                    makeDefault = {
+                        Report()
+                    },
+                    uiUpdate = { loadedReport ->
+                        _uiState.update { prev ->
+                            prev.copy(
+                                reportOptions2 = loadedReport?.reportOptions?.takeIf { it.isNotBlank() }
+                                    ?.let { Json.decodeFromString(ReportOptions2.serializer(), it) }
+                                    ?: ReportOptions2()
+                            )
+                        }
+                    }
+                )
+                navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT).collect { result ->
```

Comment:
Result returner should not be used like this. The ReportFilterEditViewModel should use receive and return only the ReportFilter2, not anything else.

Reply:
This is still an issue. There is no need to use a map. ReportFilterEditViewModel should 'return' ReportFilter3 as the result, and that should be then be collected. No need for a map (same as other screens).